frontend/src/metabase/visualizations/visualizations/WaterfallChart.jsx

```diff
@@ -53,6 +55,21 @@ export default class WaterfallChart extends LineAreaBarChart {
 
   static renderer = waterfallRenderer;
 
+  static checkRenderable = (series, settings) => {
+    super.checkRenderable(series, settings);
+
+    const [{ card }] = series;
+    const question = new Question(card);
+    // query().breakouts() does not exist in native queries
+    const queryBreakouts = question.query().breakouts?.() || [];
+
+    if (queryBreakouts.length > this.maxDimensionsSupported) {
+      throw new Error(
+        t`${this.uiName} chart does not support queries with multiple groupings`,
```

Comment:
Good catch! I think I missed this in the first review.

But nit: `${this.uiName}` suggests there might be more than one value for this. This is true for `LineAreaBarChart` https://github.com/metabase/metabase/blob/69743e83bfa4be5f442ff99f2ed6ebd2ef882f62/frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx#L99

But not for waterfall chart.

I'd just use a simple string here instead.
```suggestion
        t`Waterfall chart does not support queries with multiple groupings`,
```